### ISSUE TITLE
Deep Learning

### ISSUE THREAD

**Issue by azhe825** [azhe825/Literature-Review#13]
### Solution for overfitting:
1. Pre-training (Unsupervised) layer by layer (Greedy: RMB, autoencoder). Rather than random initialization
2. Fine-tuning (supervised)

Drop out:
each hidden unit has a 0.5 change to be set to 0. while back-propagating. In iterations, masks will change. (Train to be more robust)
After training, when testing, each hidden unit is set to be 0.5.
Drop out training can be slower.
### Solution for underfitting:

Better optimization method

Pre-train autoencoder. (For compression and reconstruction)
### Deep Belief Network

Top two layers form a RBM (undirected connections)

Other layers, Bayesian Network (directed connections)

Training is hard. Initialization is crucial.

**Comment by azhe825** [comment]
Gip sampling, RBM

**Comment by azhe825** [comment]
A fast learning algorithm for deep belief nets

**Comment by azhe825** [comment]
Activation function:
- Sigmoid: sigm(a)=1/(1+exp(-a))       0 to 1
- hyperbolic tangent: tanh(a)=(exp(2a)-1)/(exp(2a)+1)   -1 to 1
- rectified linear: reclin(a)=max(0,a)     0 to infty
